src/Microsoft.ML.Dnn/DnnCatalog.cs

```diff
@@ -85,33 +83,50 @@ public static class DnnCatalog
+        /// <param name="epoch">Number of training iterations. Each iteration/epoch refers to one pass over the dataset.</param>
         /// <param name="batchSize">The batch size for training.</param>
         /// <param name="learningRate">The learning rate for training.</param>
+        /// <param name="metricsCallback">Callback for reporting model statistics during training phase.</param>
+        /// <param name="statisticFrequency">Indicates the frequency of epochs at which to report model statistics during training phase.</param>
+        /// <param name="framework">Indicates the choice of DNN training framework. Currently only tensorflow is supported.</param>
+        /// <param name="modelSavePath">Optional name of the path where a copy new graph should be saved. The graph will be saved as part of model.</param>
+        /// <param name="checkpointName">The name of the prefix for checkpoint files.</param>
+        /// <param name="validationSet">Validation set.</param>
+        /// <param name="testOnTrainSet">Indicates to evaluate the model on train set after every epoch.</param>
+        /// <param name="reuseTrainSetBottleneckCachedValues">Indicates to not re-compute cached trainset bottleneck values if already available in the bin folder.</param>
+        /// <param name="reuseValidationSetBottleneckCachedValues">Indicates to not re-compute validataionset cached bottleneck validationset values if already available in the bin folder.</param>
+        /// <param name="trainSetBottleneckCachedValuesFilePath">Indicates the file path to store trainset bottleneck values for caching.</param>
+        /// <param name="validationSetBottleneckCachedValuesFilePath">Indicates the file path to store validationset bottleneck values for caching.</param>
         /// <remarks>
         /// The support for image classification is under preview.
         /// </remarks>
-        public static DnnEstimator ImageClassification(
+        public static ImageClassificationEstimator ImageClassification(
             this ModelOperationsCatalog catalog,
             string featuresColumnName,
             string labelColumnName,
-            string outputGraphPath = null,
             string scoreColumnName = "Score",
             string predictedLabelColumnName = "PredictedLabel",
-            string checkpointName = "_retrain_checkpoint",
-            Architecture arch = Architecture.ResnetV2101,
-            DnnFramework dnnFramework = DnnFramework.Tensorflow,
+            Architecture arch = Architecture.InceptionV3,
             int epoch = 10,
             int batchSize = 20,
-            float learningRate = 0.01f)
+            float learningRate = 0.01f,
+            ImageClassificationMetricsCallback metricsCallback = null,
+            int statisticFrequency = 1,
+            DnnFramework framework = DnnFramework.Tensorflow,
+            string modelSavePath = null,
+            string checkpointName = "_retrain_checkpoint",
```

Comment:
I think we need to change the name of the custom model generated for the user, I mean the .PB TensorFlow model file. Currently I think we have:

- `resnet_v2_101_299.meta_retrain_checkpoint.pb`
or if using Inception:
- `InceptionV3.meta_retrain_checkpoint.pb`

When the user sees those names, it'll feel like something related to an internal process (checkpoint, etc.).
Can you change it to something like the following?

- `custom.retrained.model.based.on.resnet_v2_101_299.pb`
or if using InceptionV3:
- `custom.retrained.model.based.on.InceptionV3.pb`

I'm open to comparable names, but using "checkpoint" will feel weird for a user. Feels like it is a checkpoint but still not finalized..